src/main/java/com/cloudant/client/api/ClientBuilder.java

```diff
@@ -0,0 +1,526 @@
+     * <P>
+     * Note: the supplied GsonBuilder will be augmented with some internal TypeAdapters.
+     * </P>
+     *
+     * @param gsonBuilder the custom GsonBuilder to use
+     * @return this ClientBuilder object for setting additional options
+     */
+    public ClientBuilder setGsonBuilder(GsonBuilder gsonBuilder) {
+        this.gsonBuilder = gsonBuilder;
+        return this;
+    }
+
+    /**
+     * Set the maximum number of connections to maintain in the connection pool.
+     * <P>
+     * Note: this setting only applies if using the optional OkHttp dependency. If OkHttp is not
+     * present then the JVM configuration is used for pooling. Consult the JVM documentation for
+     * the {@code http.maxConnections} property for further details.
+     * </P>
+     * Defaults to {@link #DEFAULT_MAX_CONNECTIONS}
+     *
+     * @param maxConnections the maximum number of simultaneous connections to open to the server
+     * @return this ClientBuilder object for setting additional options
+     */
+    public ClientBuilder setMaxConnections(int maxConnections) {
+        this.maxConnections = maxConnections;
+        return this;
+    }
+
+    /**
+     * Sets a proxy url for the client connection.
+     *
+     * @param proxyURL the URL of the proxy server
+     * @return this ClientBuilder object for setting additional options
+     */
+    public ClientBuilder setProxyURL(URL proxyURL) {
+        this.proxyURL = proxyURL;
+        return this;
+    }
+
+    /**
+     * Sets an optional proxy username for the client connection.
+     *
+     * @param proxyUser username for the proxy server
+     * @return this ClientBuilder object for setting additional options
+     */
+    public ClientBuilder setProxyUser(String proxyUser) {
+        this.proxyUser = proxyUser;
+        return this;
+    }
+
+    /**
+     * Sets an optional proxy password for the proxy user specified by
+     * {@link #setProxyUser(String)}.
+     *
+     * @param proxyPassword password for the proxy server user
+     * @return this ClientBuilder object for setting additional options
+     */
+    public ClientBuilder setProxyPassword(String proxyPassword) {
+        this.proxyPassword = proxyPassword;
+        return this;
+    }
+
+    /**
+     * Flag to disable hostname verification and certificate chain validation. This is not
+     * recommended, but for example could be useful for testing with a self-signed certificate.
+     * <P>
+     * The default is {@code setSSLAuthenticationDisabled(false)} meaning that hostname
+     * verification and certificate chain validation is done using the JVM default settings.
+     * </P>
+     *
+     * @param disabled true to disable, defaults to false
+     * @return this ClientBuilder object for setting additional options
+     * @see #setAuthenticatedModeSSLSocketFactory(SSLSocketFactory)
+     */
+    public ClientBuilder setSSLAuthenticationDisabled(boolean disabled) {
```

Comment:
If we go the way for removing the `set` prefix I think we should change this to `disableSSLAuthentication()`, although it would mean this builder couldn't be reused to create a more secure client by turning it back on again, I think its fine, since there isn't a way to clear the interceptors anyway. So I think the user would probably want a whole new builder so they can control which interceptors run where.